src/ManageCourses.Api/Services/TransitionService.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Linq;
+using GovUk.Education.ManageCourses.Domain.DatabaseAccess;
+
+namespace GovUk.Education.ManageCourses.Api.Services
+{
+    public class TransitionService : ITransitionService
+    {
+        private IManageCoursesDbContext _context;
+
+        public TransitionService(IManageCoursesDbContext context)
+        {
+            _context = context;
+        }
+
+        public void UpdateNewCourse(string providerCode, string courseCode, string email)
+        {
+            var optedInNewCourse = _context.GetCourse(providerCode, courseCode, email)
+                .FirstOrDefault(c =>
+                    c.Provider.OptedIn &&
+                    c.CourseSites != null &&
+                    c.CourseSites.All(cs => cs.Status.Equals("n", StringComparison.InvariantCultureIgnoreCase)));
```

Comment:
So I think this mean that if a course has a mixture of `new` and `discontinued` CourseSites, it's not gonna be set to running. This means the following test will fail:

```
Given course
with site 1 which is new
and site 2 which is suspended
when 'update new course' is called
then site 1 is running and publish='y'
and site 2 is suspended
```

@fofr do you agree with the above test?